common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -15,19 +15,28 @@
  */
 package io.netty.util.concurrent;
 
-import io.netty.util.internal.ObjectUtil;
-
-import java.util.PriorityQueue;
+import java.util.Comparator;
 import java.util.Queue;
 import java.util.concurrent.Callable;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
+import io.netty.util.internal.DefaultPriorityQueue;
+import io.netty.util.internal.ObjectUtil;
+
 /**
  * Abstract base class for {@link EventExecutor}s that want to support scheduling.
  */
 public abstract class AbstractScheduledEventExecutor extends AbstractEventExecutor {
 
+    private static final Comparator<ScheduledFutureTask<?>> SCHEDULED_FUTURE_TASK_COMPARATOR =
+            new Comparator<ScheduledFutureTask<?>>() {
+                @Override
+                public int compare(ScheduledFutureTask<?> o1, ScheduledFutureTask<?> o2) {
+                    return o1.compareTo(o2);
+                }
+            };
+
     Queue<ScheduledFutureTask<?>> scheduledTaskQueue;
```

Comment:
nit: if you use `io.netty.util.internal.PriorityQueue` then you have access to `removeTyped` which should remove some casting overhead.

Reply:
Also `clear()` is more expensive by default on `DefaultPriorityQueue` because it iterates through all tasks to reset the queue index. Consider adding another method which doesn't iterate through the collection to fix indexes and instead just sets size to `0` (e.g. `clearIgnoreIndexes()`).